orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/ArtifactResolver.java

```diff
@@ -228,14 +229,14 @@ public ArtifactResolver(
 
   public void resolveArtifacts(@Nonnull Map pipeline) {
     Map<String, Object> trigger = (Map<String, Object>) pipeline.get("trigger");
+
+    List<?> originalExpectedArtifacts =
```

Comment:
could you describe what problem you ran into here and why all the extra code was necessary to work around it - not sure I totally understand the justification just from looking at it

Reply:
This was also committed in its own commit to help document the change: 688e1f9612f9f52ab3c38d85f4feae889804c7a9 It should explain the problem at the method level.
Feel free to read these issues/PRs to better grasp the overall context: https://docs.google.com/spreadsheets/d/112ul54xZGcN5pgKfd4hJlcstDGue-0LWghyNQ8lqBEc
A better fix would require a lot more time/effort, although probably not too much for someone with deep knowlege of the codebase.

Reply:
:+1: